manifests/opsmanager/install.pp

```diff
@@ -0,0 +1,48 @@
+# PRIVATE CLASS: do not call directly
+class mongodb::opsmanager::install {
+  #assert_private("You are calling a private class mongodb::opsmanager::install.")
+  $package_ensure = $mongodb::opsmanager::package_ensure
+  $package_name   = $mongodb::opsmanager::package_name
+  $download_url   = $mongodb::opsmanager::download_url
+
+  case $package_ensure {
+    true:     {
+      $my_package_ensure = 'present'
+      $file_ensure       = 'directory'
+    }
+    false:    {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'absent': {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'purged': {
+      $my_package_ensure = 'purged'
+      $file_ensure       = 'absent'
+    }
+    default:  {
+      $my_package_ensure = $package_ensure
+      $file_ensure       = 'present'
+    }
+  }
+
+  case $facts['os']['family'] {
+    'RedHat': {
+      $my_provider = 'rpm'
```

Comment:
Maybe wrap this with `if versioncmp(fact('puppetversion'),'5.4.0') < 0`?
This would make it more obvious to rip out later when eventually we drop support for earlier puppet versions.

P.S.  Check my syntax is actually correct!